app/controllers/people/external_invoices_controller.rb

```diff
@@ -15,6 +15,12 @@ def cancel
     redirect_to external_invoices_group_person_path(group, person)
   end
 
+  def show
+    person_id = invoice.person_id
+    group_id = Person.where(id: person_id).pick(:primary_group_id)
+    redirect_to external_invoices_group_person_path(group_id, person_id)
+  end
```

Comment:
Is it OK to do this in this controller? Or should I create a separate `top` controller like it is done for https://github.com/hitobito/hitobito/blob/master/app/controllers/person/top_controller.rb for example?

Reply:
That's completely fine here. It's rather a Nebenschauplatz and not so crucial as a `Person`.